packages/lib/server/queries/teams/index.ts

```diff
@@ -294,7 +294,12 @@ export async function updateNewTeamMemberEventTypes(userId: number, teamId: numb
   });
 
   const allManagedEventTypePropsZod = _EventTypeModel.pick(allManagedEventTypeProps);
-
+  if (eventTypesToAdd.length > 0) {
+    await prisma.eventType.updateMany({
+      where: { userId, previousSlug: { in: eventTypesToAdd.map((e) => e.slug) } },
+      data: { previousSlug: null },
+    });
+  }
```

Comment:
When a new user joins the team clear the previousSlugs which are the same as the slug of event-types to add to user.